gitpod-io/gitpod
# [experiments] Add abstraction for configcat to work in self-hosted

## Description

Adds an interface in front of configcat. This allows the following:
1. Different SDK keys to be used for different environments (specified as ENV variable)
3. When not specified, uses a default implementation which always returns the default value. This is essential to not break air-gapped environments for self-hosted.

We use a shared wrapper interface for experiments implemented in `@gitpod-protocol`. We then either use a browser version of configcat client for dashboard, or `configcat-node` for backend.

## Related Issue(s)

Fixed: https://github.com/gitpod-io/gitpod/issues/10524

* Fixes air-gapped envs for self-hosted
* Depends on CONFIGCAT_SDK_KEY being added to the server pod, or any other pod that wants to use feature flags from TS on the server side - https://github.com/gitpod-io/gitpod/pull/10819

## How to test

1. Start a preview environment
2. Create team `Gitpod`
3. Try to configure usage based pricing - if you can see it, and configure it then this PR works

## Release Notes

```release-note
NONE
```

## Documentation

NONE
